common-code/azure_wrappers/azure_container.py

```diff
@@ -0,0 +1,355 @@
+        )
+    else:
+        overwrite_allow = True
+    assert overwrite_allow, "Pick a new file name and retry"
+    if "parquet" in file_name:
+        upload_parquet(data, account_url, container_name, file_name, metadata=metadata)
+        print("File successfully uploaded to", container_name, "as", file_name)
+    elif "csv" in file_name:
+        # Upload file as a csv to blob
+        container_client.upload_blob(
+            name=file_name,
+            data=data.to_csv(),
+            overwrite=True,
+            timeout=14400,
+            metadata=metadata,
+        )
+        print("File successfully uploaded to", container_name, "as", file_name)
+    elif "xlsx" in file_name or ".txt" in file_name:
+        container_client.upload_blob(
+            name=file_name, data=data, overwrite=True, timeout=14400, metadata=metadata
+        )
+        print("File successfully uploaded to", container_name, "as", file_name)
+    elif ".pdf" in file_name:
+        if debug:
+            print("uploading a pdf")
+        container_client.upload_blob(
+            name=file_name,
+            data=data,
+            overwrite=True,
+            content_settings=ContentSettings(content_type="application/pdf"),
+            timeout=14400,
+            metadata=metadata,
+        )
+        print("File successfully uploaded to", container_name, "as", file_name)
+
+    else:
+        print(
+            "Failed to upload. Please include extension in file path. \
+              The function currently supports csv and parquet."
+        )
+
+
+def upload_excel(data, account_url, container_name, file_name):
+    pass
+
+
+def upload_parquet(df, account_url, container_name, file_name, metadata=None):
+    """
+    Convert pandas dataframe to parquet and upload to Azure.
+    """
+    table = pa.Table.from_pandas(df)
+    buf = pa.BufferOutputStream()
+    pq.write_table(table, buf)
+    tenant_id = os.environ.get("TENANT_ID", DEFAULT_TENANT_ID)
+    container_client = get_container_client(
+        account_url, container_name, credential=get_credential(tenant_id)
+    )
+
+    container_client.upload_blob(
+        name=file_name,
+        data=buf.getvalue().to_pybytes(),
+        overwrite=True,
+        timeout=14400,
+        metadata=metadata,
+    )
+
+
+def upload_files_from_folder(
+    account_url, container_name, source_folder_path, dest_folder_name=None
+):
+    all_file_names = os.listdir(source_folder_path)
+    for file_name in all_file_names:
+        upload_file_from_path(
+            account_url,
+            container_name,
+            source_folder_path,
+            file_name,
+            dest_folder_name=dest_folder_name,
+        )
+
+
+def upload_file_from_path(
+    account_url,
+    container_name,
+    source_folder_path,
+    file_name,
+    dest_folder_name=None,
+    metadata=None,
+):
+    """
+    Uploads a file to a container from a given filepath.
+    Currently only supports PDFs.
```

Comment:
I think the "currently only supports PDFs" is no longer true.

Reply:
```suggestion
```